bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/DocumentCloudViewerImpl.java

```diff
@@ -0,0 +1,234 @@
+
+@Model(adaptables = SlingHttpServletRequest.class, adapters = { DocumentCloudViewer.class,
+        ComponentExporter.class }, resourceType = { DocumentCloudViewerImpl.RESOURCE_TYPE })
+@Exporter(name = ExporterConstants.SLING_MODEL_EXPORTER_NAME, extensions = ExporterConstants.SLING_MODEL_EXTENSION)
+public class DocumentCloudViewerImpl implements DocumentCloudViewer {
+
+    protected static final String RESOURCE_TYPE = "core/wcm/components/documentcloudviewer/v1/documentcloudviewer";
+    protected static final String FULL_WINDOW = "FULL_WINDOW";
+    protected static final String SIZED_CONTAINER = "SIZED_CONTAINER";
+    protected static final String IN_LINE = "IN_LINE";
+    protected static final String EMBED_MODE = "embedMode";
+    protected static final String DEFAULT_VIEW_MODE= "defaultViewMode";
+    protected static final String SHOW_ANNOTATION_TOOLS = "showAnnotationTools";
+    protected static final String SHOW_LEFT_HAND_PANEL = "showLeftHandPanel";
+    protected static final String SHOW_FULL_SCREEN = "showFullScreen";
+    protected static final String SHOW_PAGE_CONTROLS = "showPageControls";
+    protected static final String DOCK_PAGE_CONTROLS = "dockPageControls";
+    protected static final String SHOW_DOWNLOAD_PDF = "showDownloadPDF";
+    protected static final String SHOW_PRINT_PDF = "showPrintPDF";
+    protected static final String CSS_FULL_WINDOW = "cmp-documentcloudviewer__full-window";
+    protected static final String CSS_BORDERLESS = "cmp-documentcloudviewer__full-window-borderless";
+    protected static final String CSS_SIZED_CONTAINER = "cmp-documentcloudviewer__sized-container";
+    protected static final String CSS_IN_LINE = "cmp-documentcloudviewer__in-line";
+
+    @ValueMapValue(injectionStrategy = InjectionStrategy.OPTIONAL)
+    private String documentPath;
+
+    @ValueMapValue(injectionStrategy = InjectionStrategy.OPTIONAL)
+    private String type;
+
+    @ValueMapValue(injectionStrategy = InjectionStrategy.OPTIONAL)
+    private String defaultViewMode;
+
+    @ValueMapValue(injectionStrategy = InjectionStrategy.OPTIONAL)
+    private String viewerHeight;
+
+    @ValueMapValue(injectionStrategy = InjectionStrategy.OPTIONAL)
+    private boolean borderless;
+
+    @ValueMapValue(injectionStrategy = InjectionStrategy.OPTIONAL)
+    private boolean showAnnotationTools;
+
+    @ValueMapValue(injectionStrategy = InjectionStrategy.OPTIONAL)
+    private boolean showFullScreen;
+
+    @ValueMapValue(injectionStrategy = InjectionStrategy.OPTIONAL)
+    private boolean showLeftHandPanel;
+
+    @ValueMapValue(injectionStrategy = InjectionStrategy.OPTIONAL)
+    private boolean showDownloadPdf;
+
+    @ValueMapValue(injectionStrategy = InjectionStrategy.OPTIONAL)
+    private boolean showPrintPdf;
+
+    @ValueMapValue(injectionStrategy = InjectionStrategy.OPTIONAL)
+    private boolean showPageControls;
+
+    @ValueMapValue(injectionStrategy = InjectionStrategy.OPTIONAL)
```

Comment:
apart from `resource` all other fields seem to be optional, it might make sense to make `DefaultInjectionStrategy` to Optional inside `@Model` annotation and add `@Required` to `resource` field